jiboia/core/service/project_svc.py

```diff
@@ -0,0 +1,131 @@
+import logging
+from collections import Counter
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+from django.db.models import Sum
+
+from ..models import Issue, Project, StatusLog, TimeLog
+
+logger = logging.getLogger(__name__)
+
+def group_issues_per_month(issues, start_date, end_date):
+    results = []
+    current = start_date
+
+    while current <= end_date:
+        month_issues = issues.filter(
+            created_at__year=current.year,
+            created_at__month=current.month,
+        )
+
+        counter = Counter({
+            "pending": 0,
+            "on_going": 0,
+            "mr": 0,
+            "concluded": 0,
+        })
+
+        for issue in month_issues:
+            last_log = (
+                issue.statuslog_set.order_by("-created_at")
+                .select_related("new_status")
+                .first()
+            )
+            if last_log and last_log.new_status:
+                status_name = last_log.new_status.name.lower()
+                if status_name in counter:
+                    counter[status_name] += 1
+
+        results.append({
+            "date": current.strftime("%m/%Y"),
+            "pending": counter["pending"],
+            "on_going": counter["on_going"],
+            "mr": counter["mr"],
+            "concluded": counter["concluded"],
+        })
+
+        current += relativedelta(months=1)
+
+    return results
+
+
+def list_projects_general(issue_breakdown_months: int):
```

Comment:
Está muito grande as responsabilidade dessa função, ela consegue ser quebrada em modulos menores e mais reutilizáveis. Está rolando iteração com banco na camada de service, isso nao é legal, manda para camada de model. Os for independetes denunciam que podiam ser duas funções distintas, talvez. 
Com essas funções menores, haverá mais testes, como também fica mais fácil de testar